### ISSUE TITLE
[Feature] allow multiple jobs from one recurring job in parallel

### ISSUE THREAD

**Issue by uzaironline** [jobrunr/jobrunr#398]
Are you running the latest version of JobRunr?
Yes
5.0.0-RC1

Next job run does not trigger until previous execution finishes.

I'm using JobRunr version: 5.0.0-RC1
I'm running on jdk1.8.0_131
I'm using StorageProvider as InMemoryStorageProvider 

Provide a simple Github project or code fragment which shows how to reproduce the issue (ideally a Github repo).
```java
public class ReportApplication {
    public static void main(String[] args) throws InterruptedException {
        configure();
        BackgroundJob.scheduleRecurrently(Cron.minutely(), ()-> {
            doWorkThatTakesLongInterruptThread(120);
        });
        Thread.currentThread().join();
    }

    public static void doWorkThatTakesLongInterruptThread(int seconds) {
        try {
            System.out.println("WORK IS STARTED!!!!!!!!" + Instant.now());
            TimeUnit.SECONDS.sleep(seconds);
            System.out.println("WORK IS DONE!!!!!!!!" + Instant.now());
        } catch (InterruptedException e) {
            System.out.println("Thread has been interrupted");
            Thread.currentThread().interrupt();
        }
    }

    private static void configure() {
        JobRunr.configure()
                .useStorageProvider(new InMemoryStorageProvider())
                .useBackgroundJobServer()
                .useDashboard()
                .initialize();

    }
}

logs..
WORK IS STARTED!!!!!!!!2022-03-29T17:44:00.140Z
WORK IS DONE!!!!!!!!2022-03-29T17:46:00.152Z
WORK IS STARTED!!!!!!!!2022-03-29T17:46:47.147Z

```

**Expected behavior**
The Job should be triggered recurring at the time when it was scheduled. In crontab jobs are triggered as per scheduled, it does not wait for previous execution to complete.

**Comment by rdehuyss** [comment]
That's a nice one - I however decided not to implement it like this (and also others have requested to not do it). 

Reason is that your server then can go out of resources (because creating jobs is happening faster than processing them).

Is there any reason why you find this important?

**Comment by uzaironline** [comment]
This is always not the case! We have few reports which are triggered at every 15 minute interval. In some cases the execution time is more that 15 minutes for some reason. The report captures time sensitive information. Start and End time is calculated based on the job trigger time. 
Every 15 minute job capture the data that has been modified in past 15 minutes. 
e.g. 
1st trigger
trigger-time - 30-Mar-2022T01:00:00 then start and end time determined as follows
start-time = trigger-time - 15min = 30-Mar-2022T00:45:00
end-time = trigger-time = 30-Mar-2022T01:00:00

if for some reason first execution runs longer and finishes in 18minutes, then the second run will be delayed by 3mins and it will also slide the start and end time calculation. This will loose to capture any modification in 3min delay. 
2nd trigger
trigger-time - 30-Mar-2022T01:18:00 
start-time = trigger-time - 15min = 30-Mar-2022T01:03:00
end-time = trigger-time = 30-Mar-2022T01:18:00

The delay is not a problem but unable to determine start and end time correctly in case of delay.

**Comment by rdehuyss** [comment]
I'm open for a PR where as a developer you could decide on either the current way (e.g;. wait for the previous job to finish) or your preferred way where multiple jobs can run in parallel.

**Comment by rdehuyss** [comment]
Solved in JobRunr Pro 6.0.
